system/mp/mpbase.cpp

```diff
@@ -707,6 +708,25 @@ IGroup *createIGroup(const char *endpointlist,unsigned short defport)
     return CGroup::fromText(endpointlist,defport);
 }
 
+IGroup *createIGroupRetry(const char *endpointlist,unsigned short defport, unsigned timeout)
+{
+    CTimeMon t(timeout);
+    while (!t.timedout())
+    {
+        try
+        {
+            return createIGroup(endpointlist, defport);
+        }
+        catch (IException *e)
+        {
+            VStringBuffer errMsg("Failed to resolve group for: %s", endpointlist);
+            EXCLOG(e, errMsg.str());
+            e->Release();
+        }
```

Comment:
Should we add a sleep()  or delay here to avoid a heavy spin ?

Reply:
yes, I think worth adding a pause here.